app/javascript/packs/webauthn-authenticate.js

```diff
@@ -1,34 +1,59 @@
 const WebAuthn = require('../app/webauthn');
 
 function webauthn() {
-  // If webauthn is not supported redirect back to the 2fa options list
   const webauthnInProgressContainer = document.getElementById('webauthn-auth-in-progress');
   const webauthnSuccessContainer = document.getElementById('webauthn-auth-successful');
 
-  if (!WebAuthn.isWebAuthnEnabled()) {
-    const href = webauthnInProgressContainer.getAttribute('data-webauthn-not-enabled-url');
-    window.location.href = href;
-  }
+  const webauthnPlatformRequested =
+    webauthnInProgressContainer.dataset.platformAuthenticatorRequested === 'true';
+  const multipleFactorsEnabled =
+    webauthnInProgressContainer.dataset.multipleFactorsEnabled === 'true';
+  const isPlatformAvailable = document.getElementById('webauthn_device').value === 'true';
 
   const spinner = document.getElementById('spinner');
   spinner.classList.remove('hidden');
 
-  WebAuthn.verifyWebauthnDevice({
-    userChallenge: document.getElementById('user_challenge').value,
-    credentialIds: document.getElementById('credential_ids').value,
-  }).then((result) => {
-    document.getElementById('credential_id').value = result.credentialId;
-    document.getElementById('authenticator_data').value = result.authenticatorData;
-    document.getElementById('client_data_json').value = result.clientDataJSON;
-    document.getElementById('signature').value = result.signature;
-    webauthnInProgressContainer.classList.add('hidden');
-    webauthnSuccessContainer.classList.remove('hidden');
-  });
+  if (
+    !WebAuthn.isWebAuthnEnabled() ||
+    (webauthnPlatformRequested && !isPlatformAvailable && !multipleFactorsEnabled)
+  ) {
+    const href = webauthnInProgressContainer.getAttribute('data-webauthn-not-enabled-url');
+    window.location.href = href;
+  } else {
+    // if platform auth is not supported on device, we should take user to the error screen if theres no additional methods.
+    // if platform auth is not supported on device, we should take user to the error screen if theres no additional methods.
```

Comment:
whoops, I bet a merge conflict resulted in us duplicating this line like this, luckily it's a comment so it's a no-op

Reply:
> whoops, I bet a merge conflict resulted in us duplicating this line like this, luckily it's a comment so it's a no-op

👍  Related: https://github.com/18F/identity-idp/pull/5976#discussion_r824885345